app/src/main/java/com/github/se/polyfit/ui/components/selector/MealSelector.kt

```diff
@@ -0,0 +1,162 @@
+                tint = SecondaryGrey,
+                modifier = Modifier.padding(end = 20.dp))
+
+            if (!isDisplayMealDetails) {
+              Text(
+                  text = context.getString(R.string.shareRecipePlaceholder),
+                  color = SecondaryGrey,
+                  fontSize = 20.sp,
+              )
+            } else {
+              Text(
+                  text = selectedMeal.name,
+                  fontSize = 20.sp,
+                  modifier = Modifier.weight(1f),
+              )
+              Text(
+                  text = "${selectedMeal.calculateTotalCalories()} kcal",
+                  fontSize = 20.sp,
+                  modifier = Modifier.weight(1f).padding(end = 20.dp),
+                  textAlign = TextAlign.End,
+              )
+            }
+          }
+    } else {
+      MealInputTextField(
+          meals = meals,
+          onCollapseSearchBar = { showMealSearch = false },
+          onMealChange = {
+            setSelectedMeal(it)
+            setPostMeal(it)
+          })
+    }
+    if (isDisplayMealDetails) {
+      MealDetails(selectedMeal, Modifier.padding(top = 10.dp))
+    }
+  }
+}
+
+@Composable
+fun MealDetails(meal: Meal, modifier: Modifier = Modifier) {
+  Box(modifier = modifier.fillMaxWidth().testTag("MealDetails")) {
+    Column(modifier = Modifier.fillMaxWidth(0.75f).align(Alignment.Center)) {
+      LazyColumn(modifier = Modifier.heightIn(max = 200.dp)) {
+        items(meal.ingredients) { ingredient ->
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(4.dp).testTag("Ingredient"),
+              verticalAlignment = Alignment.CenterVertically) {
+                Text(
+                    text = ingredient.name,
+                    modifier = Modifier.weight(1f).padding(start = 10.dp),
+                    fontSize = 16.sp,
+                    color = SecondaryGrey)
+
+                Text(
+                    text = "${ingredient.amount} ${ingredient.unit.toString().lowercase()}",
+                    modifier = Modifier.weight(1f).padding(end = 10.dp),
+                    textAlign = TextAlign.End,
+                    fontSize = 16.sp,
+                    color = SecondaryGrey)
+              }
+          HorizontalDivider(color = Color.LightGray)
+        }
+      }
+
+      Spacer(modifier = Modifier.height(10.dp))
+
+      Row(modifier = Modifier.align(Alignment.CenterHorizontally)) {
+        val macros = meal.getMacros()
+        val nutrients =
+            listOf(
+                Pair("Carbs", macros["carbohydrates"]),
+                Pair("Fat", macros["fat"]),
+                Pair("Protein", macros["protein"]),
+            )
+
+        nutrients.forEach { (nutrientName, amount) ->
+          GradientBox(
+              outerModifier = Modifier.padding(horizontal = 10.dp),
+              innerModifier =
+                  Modifier.heightIn(min = 80.dp, max = 80.dp).widthIn(max = 80.dp, min = 80.dp),
```

Comment:
if the range is not actually a difference, why not just use 
```suggestion
                  Modifier.height(80.dp).width(80.dp),
```